### ISSUE TITLE
Missing Test Case - 1458. Max Dot Product of Two Subsequences

### ISSUE THREAD

**Issue by shivansh-yashasvi** [LeetCode-Feedback/LeetCode-Feedback#17102]
#### Your LeetCode username

yashasvi_shivansh_07

#### Category of the bug
- [ ] Question
- [ ] Solution
- [ ] Language
- [x] Missing Test Cases 

#### Description of the bug

My code is wrong with the test case below but got accepted. 
The submission is https://leetcode.com/submissions/detail/1072217248/

The test case which is missing is 
``` 
nums1 = [-1,-1]
```
```
nums2 = [-2,-3,-4]
```

All test cases in which both nums1 and nums2 contain all negative numbers will be in the class of missing testcases as my accepted solution will fail all of them. This is because the following codepiece is incorrect:
```
// incorrect code piece
if(nums1max < 0 || nums2max < 0) {
    return max(nums1max*nums2min , nums1min*nums2max);
}
```
The correct codepiece would have been:
```
// correct code piece
if( (nums1max < 0 && nums2min >= 0) || (nums2max < 0 && nums1min >=0)) {
    return max(nums1max*nums2min , nums1min*nums2max);
}
```

#### Code you used for Submit/Run operation

```
// 1458. Max Dot Product of Two Subsequences
class Solution {
public:
    int maxDotProduct(vector<int>& nums1, vector<int>& nums2) {
        ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

        int nums1min = *min_element(nums1.begin(), nums1.end());
        int nums1max = *max_element(nums1.begin(), nums1.end());

        int nums2min = *min_element(nums2.begin(), nums2.end());
        int nums2max = *max_element(nums2.begin(), nums2.end());

        if(nums1max < 0 || nums2max < 0) {
            return max(nums1max*nums2min , nums1min*nums2max);
        }

        vector<vector<int>> memo(nums1.size()+1, vector<int>(nums2.size()+1, -1));
        return helper(nums1, nums2, 0, 0, memo);
    }

    int helper(vector<int>& nums1, vector<int>& nums2, int i, int j, vector<vector<int>>& memo) {
        if( i == nums1.size() || j == nums2.size() ) return 0;

        if(memo[i][j] != -1) return memo[i][j];

        // take both 
        int takeboth = nums1[i]*nums2[j] + helper(nums1, nums2, i+1, j+1, memo);

        // dont take from nums2 
        int takenum1 = helper(nums1, nums2, i, j+1, memo);

        // dont take from nums1
        int takenum2 = helper(nums1, nums2, i+1, j, memo);

        return memo[i][j] = max({takeboth, takenum1, takenum2});
    }
};
```

#### Language used for code

C++

#### Expected behavior

For the testcase given below,
``` 
nums1 = [-1,-1]
```
```
nums2 = [-2,-3,-4]
```
The correct output should be:
```
Expected Output = 7 
```

But my actual output was:
```
Actual Output = 4
```

I expected that the submitted code would not be accepted.  But the code was accepted despite it being wrong. Hence, additional missing testcases should be added to address this issue.

#### Screenshots

This is the testcase which needs to be added. My code gives wrong output for the same when I do custom testing.
<img width="1440" alt="Screenshot 2023-10-11 at 5 18 46 AM" src="https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/83114748/26db8e02-bd88-453c-896d-f172ee20efa1">

The same code is accepted. 
<img width="1440" alt="Screenshot 2023-10-11 at 5 19 05 AM" src="https://github.com/LeetCode-Feedback/LeetCode-Feedback/assets/83114748/09fc8b8e-d25d-4a17-8535-48fe8158f097">

**Comment by LC-Pam** [comment]
Thank you for reaching out to us. To help us further assess this issue and reproduce this issue on our end, would you please share the following information with us:

The code AC submission link. 
The above code received the wrong answer after submission: https://leetcode.com/submissions/detail/1072738609/

We appreciate your time and support. If you have any other questions, please don't hesitate to let us know.

**Comment by shivansh-yashasvi** [comment]
@LC-Pam , please provide the updates. 
I have given my AC submission link : https://leetcode.com/submissions/detail/1072217248/

**Comment by LC-Pam** [comment]
Dear user, 
 
Thank you for submitting your code to our platform. After a thorough evaluation, we regret to inform you that your code did not produce the expected results. Our code judge is functioning correctly, and we have verified the correctness of our evaluation process.
 
We kindly suggest that you review your code carefully and ensure it adheres to the problem requirements and constraints. It may be beneficial to double-check your logic, algorithm implementation, and any potential errors or corner cases that could affect the code's performance.

The above code currently received the Wrong Answer after submit: https://leetcode.com/submissions/detail/1074490525/
 
Additionally, we encourage you to explore our discussion board, where you can seek assistance from the community. Many experienced users and experts are available to help troubleshoot issues, provide insights, and offer guidance on improving your code.
 
We believe that by reviewing your code and actively engaging with the community, you will have a better chance of identifying and resolving the discrepancies. Remember, learning from such experiences and iterating on your code is an essential part of the programming journey.
 
We appreciate your understanding and encourage you to continue learning and improving your coding skills. 
 
Should you have any further questions or need additional support, please don't hesitate to reach out to us.
 
Best regards,
 
The Leetcode Team

**Comment by shivansh-yashasvi** [comment]
@LC-Pam , The testcases have only been updated after I raised this issue. The failed submission link that you provided fails on the 64th test case. While my AC submission ( link : https://leetcode.com/submissions/detail/1072217248/ ) has only 62 testcases. I'd urge you to open the link I provided and verify it for yourself as the same code has an "Accepted" verdict. 
It took me quite some time and effort to figure out this missing test case and I'd request you to kindly award the leetcoins if possible.
